backend/resources/participant.py

```diff
@@ -55,4 +56,19 @@ def delete(self, id):
         db.session.commit()
 
         return {"message": "Participant deleted successfully."}, HTTPStatus.OK
-    
\ No newline at end of file
+
+    @jwt_required
+    def put(self, id):
+        participant_schema = ParticipantSchema()
+        participant = Participant.query.get_or_404(id)
+        json_data = request.get_json(force=True)
+        try:
+            data = participant_schema.load(json_data)
+        except ValidationError as err:
+            return (err.messages), HTTPStatus.BAD_REQUEST
+
+        for key, value in data.items():
+            setattr(participant, key, value)
```

Comment:
Also from Marshmallow docs - they use `make_object` with `@post_load` decorator to create DB object just by `participant = participant_schema.load(json_data)`. What do you think about that?

see: https://marshmallow.readthedocs.io/en/stable/examples.html#todo-api-flask-peewee
or here: https://github.com/marshmallow-code/marshmallow-sqlalchemy/blob/dev/src/marshmallow_sqlalchemy/schema.py#L183

Reply:
BTW is there any reason that we settled with `marshmallow` and not `marshmallow-sqlalchemy`? it looks like it could help us here a lot.

Here are a few questions that should be somehow addressed:
* should that endpoint update all fields in participant model? (meaning that if there's no key in json payload, it shoulld set that value to null?
* I understand the necessity of having the first name required, but if we will settle with partial updates (meaning that existing instance will be a base that we will update on) so do we really need to provide this particular field every time we want to update this participant (even if the first name will not change)?